src/components/mutation/common.rs

```diff
@@ -0,0 +1,546 @@
+//! Common mutation components.
+
+use std::marker::PhantomData;
+
+use better_any::{Tid, TidAble};
+use derive_more::{Deref, DerefMut};
+use eyre::{ensure, WrapErr};
+use itertools::multizip;
+use rand::{
+    distributions::{Distribution, Uniform},
+    seq::{IteratorRandom, SliceRandom},
+    Rng,
+};
+use rand_distr::Normal;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    component::{AnyComponent, ExecResult},
+    components::{mutation::functional as f, Component},
+    identifier::{Global, Identifier},
+    population::AsSolutionsMut,
+    problems::{LimitedVectorProblem, VectorProblem},
+    CustomState, State,
+};
+
+#[derive(Deref, DerefMut, Tid)]
+pub struct MutationStrength<T: AnyComponent + 'static>(
+    #[deref]
+    #[deref_mut]
+    f64,
+    PhantomData<T>,
+);
+
+impl<T: AnyComponent> MutationStrength<T> {
+    pub fn new(value: f64) -> Self {
+        Self(value, PhantomData)
+    }
+}
+
+impl<T: AnyComponent> CustomState<'_> for MutationStrength<T> {}
+
+#[derive(Deref, DerefMut, Tid)]
+pub struct MutationRate<T: AnyComponent + 'static>(
+    #[deref]
+    #[deref_mut]
+    f64,
+    PhantomData<T>,
+);
+
+impl<T: AnyComponent> MutationRate<T> {
+    pub fn new(value: f64) -> Self {
+        Self(value, PhantomData)
+    }
+
+    pub fn value(&self) -> ExecResult<f64> {
+        ensure!(
+            (0.0..1.0).contains(&self.0),
+            "mutation rate must be in [0, 1)"
+        );
+        Ok(self.0)
+    }
+}
```

Comment:
Why can't I have a mutation rate of 1?

Reply:
That is most certainly a copy-paste error.